optimum/executorch/modeling.py

```
@@ -382,8 +384,7 @@ def generate(
             )
             max_seq_len = self.max_cache_size
 
-        # Initialize with start token (0 for T5)
-        decoder_input_ids = torch.tensor([[0]], dtype=torch.long)
+        decoder_input_ids = torch.tensor([[self.decoder_start_token_id]], dtype=torch.long)
```

@michaelbenayoun FYI, addressed your comment in https://github.com/huggingface/optimum-executorch/pull/36

---

It seems that `self.decoder_start_token_id` exists only if `"decoder_start_token_id" in metadata`.
Should we define a default value for this attribute?
Otherwise we would need to check that `self` actually has this attribute in the forward or it could fail, right?

---

Yeah, the metadata must be recorded to the PTE file to instruct the runtime for inference.So for encoder-decoder model which the `decoder_start_token_id` field is mandatory, if it's missing, we should raise an error.